BaseballPlayer.h

```diff
@@ -3,17 +3,25 @@
 #include "Human.h"
 #include <memory>
 
+class BaseballMitt;
+
 class BaseballPlayer : public Human {
 
     public:
         BaseballPlayer() = default;
         explicit BaseballPlayer(std::string a_name);
         ~BaseballPlayer();
 
-        const std::string& get_name() const;
+        const std::string & get_name() const;
+        void set_name(std::string name);
+
         bool has_baseball() const;
         void catch_baseball();
         void throw_baseball(BaseballPlayer &catch_partner);
+        void put_on_mitt(BaseballMitt mitt);
+        const BaseballMitt & get_mitt();
+
+        static inline unsigned num_catches;
```

Comment:
Why static? You cant to count total number of catches across all players?

Are there any threading issues here?
(Hint: probably)

Better might be to give each player his own count, and then main() would add them all together for the total

Reply:
I was honestly trying to force the use of a static member. Really was not thinking "Oh this is the best approach to use a static member" Sort of forcing the issue here for practice sake. It should not have a threading issue because only one player can have the baseball at any given time and each of the threads has to lock the shared mutex in order to make changes regardless. I agree not the best design just forcing the issue.